include/seqan3/range/view/take_line.hpp

```diff
@@ -0,0 +1,469 @@
+        using reference             = typename std::iterator_traits<base_base_t>::reference;
+        using pointer               = typename std::iterator_traits<base_base_t>::pointer;
+        using iterator_category     = typename std::iterator_traits<base_base_t>::iterator_category;
+        //!\}
+
+        /*!\name Arithmetic operators
+         * \brief seqan3::detail::inherited_iterator_base operators are used unless specialised here.
+         * \{
+         */
+        iterator_type & operator++()
+        {
+            *this = ++static_cast<base_base_t>(*this);
+
+            if constexpr (!forward_range_concept<urng_t>) // consuming behaviour for input ranges
+            {
+                if (!at_end)
+                {
+                    if (**this == '\r')
+                    {
+                        *this = ++static_cast<base_base_t>(*this);
+                        at_end = true;
+                    }
+
+                    if (**this == '\n')
+                    {
+                        *this = ++static_cast<base_base_t>(*this);
+                        at_end = true;
+                    }
+                }
+            }
+
+            return *this;
+        }
+
+        iterator_type operator++(int)
+        {
+            iterator_type cpy{*this};
+            ++(*this);
+            return cpy;
+        }
+        //!\}
+
+        /*!\name Comparison operators
+         * \brief We define comparison against self and against the sentinel.
+         * \{
+         */
+        bool operator==(iterator_type const & rhs) const noexcept(!require_eol)
+        {
+            return static_cast<base_base_t>(*this) == static_cast<base_base_t>(rhs);
+        }
+
+        bool operator==(sentinel_type const & rhs) const noexcept(!require_eol)
+        {
+
+            if constexpr (!forward_range_concept<urng_t>)
+            {
+                if (at_end)
+                    return true;
+            }
+
+            if (static_cast<base_base_t>(*this) == rhs)
+            {
+                if constexpr (require_eol)
+                    throw unexpected_end_of_input{"Reached end of input before end-of-line."};
+                else
+                    return true;
+            }
+
+            return **this == '\r' || **this == '\n';
+        }
+
+        friend operator==(sentinel_type const & lhs, iterator_type const & rhs) noexcept(!require_eol)
+        {
+            return rhs == lhs;
+        }
+
+        bool operator!=(sentinel_type const & rhs) const noexcept(!require_eol)
+        {
+            return !(*this == rhs);
+        }
+
+        bool operator!=(iterator_type const & rhs) const noexcept(!require_eol)
+        {
+            return static_cast<base_base_t>(*this) != static_cast<base_base_t>(rhs);
+        }
+
+        friend operator!=(sentinel_type const & lhs, iterator_type const & rhs) noexcept(!require_eol)
+        {
+            return rhs != lhs;
+        }
+        //!\}
+    };
```

Comment:
comment that this is the end of the inner class (I was wondering why there is `public:` after `public:`)

Reply:
well that's what indention is for, right :laughing: